runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -384,6 +384,8 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
       //
       clientSession->getSequencingMonitor()->enter();
       clientSession->updateMaxReceivedSeqNo(seqNo);
+      clientSession->incNumActiveThreads();
```

Comment:
Isn't this too early to increase the number of active threads? A thread that has to wait for its turn will see itself as an active thread and never clear the caches. In my mind the increment should be done after the thread has cleared the `waitForMyTurn()` routine.

Reply:
Yes, uou're right. It is too early. It should've been done right before the thread starts to do the class unloading etc as below.

```c++
      clientSession->getSequencingMonitor()->enter();
      clientSession->updateMaxReceivedSeqNo(seqNo);

      if (seqNo > clientSession->getExpectedSeqNo()) // Out of order messages
         {
         ...
         waitForMyTurn(clientSession, entry);
         }
      else if (seqNo < clientSession->getExpectedSeqNo())
         {
         ...
         clientSession->getSequencingMonitor()->exit();
         throw JITServer::StreamOOO();
         }

      clientSession->incNumActiveThreads(); <= Should be here 

      // We can release the sequencing monitor now because nobody with a
      // larger sequence number can pass until I increment expectedSeqNo
      clientSession->getSequencingMonitor()->exit();

      // At this point I know that all preceeding requests have been processed
      // and only one thread can ever be present in this section
      if (!clientSession->cachesAreCleared())
```